optimade/filtertransformers/base_transformer.py

```diff
@@ -22,29 +67,101 @@ class BaseTransformer(abc.ABC, Transformer):
         "=": None,
     }
 
-    # map from back-end specific operators to their inverse
-    # e.g. {"$lt": "$gt"} for MongoDB.
-    reversed_operator_map: Dict[str, str] = {}
+    # map from operators to their syntactic (NOT logical) inverse to handle
+    # equivalence between cases like "A > 3" and "3 < A".
+    _reversed_operator_map = {
+        ">": "<",
+        ">=": "<=",
+        "<": ">",
+        "<=": ">=",
+        "=": "=",
+        "!=": "!=",
+    }
+
+    __quantity_type: Type[Quantity] = Quantity
+    __quantities = None
 
-    def __init__(self, mapper: BaseResourceMapper = None):
+    def __init__(
+        self, mapper: BaseResourceMapper = None
+    ):  # pylint: disable=super-init-not-called
         """Initialise the transformer object, optionally loading in a
         resource mapper for use when post-processing.
 
         """
         self.mapper = mapper
 
-    def postprocess(self, query):
+    @property
+    def backend_mapping(self) -> Dict[str, Quantity]:
+        """A mapping between backend field names (aliases) and the corresponding
+        [`Quantity`][optimade.filtertransformers.base_transformer.Quantity] object.
+        """
+        return {
+            quantity.backend_field: quantity for _, quantity in self.quantities.items()
+        }
+
+    @property
+    def quantities(self) -> Dict[str, Quantity]:
+        """A mapping from the OPTIMADE field name to the corresponding
+        [`Quantity`][optimade.filtertransformers.base_transformer.Quantity] objects.
+        """
+        if self.__quantities is None:
+            self.__quantities = self.__build_quantities()
+
+        return self.__quantities
+
+    @quantities.setter
+    def __set_quantities(self, quantities: Dict[str, Quantity]) -> None:
+        self.__quantities = quantities
+
+    def __build_quantities(self) -> Dict[str, Quantity]:
```

Comment:
You are probably right (as I mentioned on Slack) - I basically did not want to break any existing ES implementations.

Before this PR, the mappers only knew about fields which had defined aliases, hence the need to pull all other fields out of the schemas and models. If we are comfortable making breaking changes then we can move some of this into the mappers, for sure.

Reply:
I can only speak for our ES implementation. Such changes would definitely break our implementation. But since, I was proposing this initial "bad" design and only did this because it was the easiest solution at the time, you should not feel too bad about breaking it. It might be good for us to get forced into provided a proper ResourceMapper in the long run.